transforms/code/header_cleanser/python/src/header_cleanser_transform.py

```diff
@@ -0,0 +1,221 @@
+                line for i, line in enumerate(code.split('\n'), 0)
+                if i not in ignore_lines
+            ])
+        return modified_code,ignore_lines!=[]
+    else :
+        return code,False
+
+def remove_license(code):
+    """
+    Using scancode.api function to detecte and remove license.
+    """
+    file_path = file_generate(content=code)
+    license_dict = api.get_licenses(file_path)
+    os.remove(file_path)
+    ignore_lines = fetch_index(license_dict)
+    if ignore_lines != []:
+        modified_code = '\n'.join([
+                line for i, line in enumerate(code.split('\n'), 0)
+                if i not in ignore_lines
+            ])
+        return modified_code,ignore_lines!=[]
+    else :
+        return code,False
+
+def remove_license_copyright(code):
+
+    file_path = file_generate(code)
+    copyright_dict = api.get_copyrights(file_path)
+    license_dict = api.get_licenses(file_path)
+    os.remove(file_path)
+    ignore_lines_license = fetch_index(license_dict)
+    ignore_lines_copyright = fetch_index(copyright_dict)
+    ignore_lines = ignore_lines_license + ignore_lines_copyright
+    if ignore_lines != []:
+        ignore_lines = check_empty_comment(code,ignore_lines)
+        modified_code = '\n'.join([
+                line for i, line in enumerate(code.split('\n'), 0)
+                if i not in ignore_lines
+            ])
+        return modified_code,True
+    else:
+        return code,False
+
+
+
+class HeaderCleanserTransform(AbstractTableTransform):
+    def __init__(self, config: dict):
+        super().__init__(config)
+
+        self.column_name = config.get(COLUMN_KEY,DEFAULT_COLUMN)
+        self.license_remove = config.get(LICENSE_KEY,DEFAULT_LICENSE)
+        self.copyright_remove = config.get(COPYRIGHT_KEY,DEFAULT_COPYRIGHT)
+
+    def transform(self, table: pa.Table,file_name: str = None) -> tuple[list[pa.Table], dict]:
+
+        contents = table.column(self.column_name).to_pylist()
+        updated_content = []
+        remove_code_count = 0
+        for content in contents:
+            if self.license_remove=="true" and self.copyright_remove=="true":
+                new_content,detect = remove_license_copyright(content)
+                if detect:
+                    remove_code_count+=1
+                updated_content.append(new_content)
+
+            elif self.copyright_remove=="true" and self.license_remove!="true":
+                new_content,detect = remove_copyright(content)
+                if detect:
+                    remove_code_count+=1
+                updated_content.append(new_content)
+
+            elif self.license_remove=='true' and self.copyright_remove!="true":
+                new_content,detect = remove_license(content)
+                if detect:
+                    remove_code_count+=1
+                updated_content.append(new_content)
+
+            else : 
+                return [table],{'Removed code count' : remove_code_count}
+
```

Comment:
The logic above does not seem correct. Can you, please, double check

Reply:
I think this works fine.
I have provided value of copyright and license as empty string and this is output of file.
"python header_cleanser_local.py 
input table has 10 rows

output table has 10 rows
Output metadata : {'Removed code count': 0}"
You can check it by runnig header_cleanser_local.py.

Reply:
It does not. Your initial test:
```
self.license_remove=="true" and self.copyright_remove=="true":
```
and your second and third tests are identical. I think the second one should be:
```
elif self.copyright_remove=="true":
```
and the third one:
```
elif self.license_remove=='true':
```

Reply:
updated. please check